django/core/handlers/wsgi.py

```diff
@@ -89,6 +96,13 @@ def __init__(self, environ):
         self._read_started = False
         self.resolver_match = None
 
+    def __getstate__(self):
+        obj_dict = self.__dict__.copy()
+        for attr in self.wsgi_non_picklable_attrs:
+            if attr in obj_dict["META"]:
+                del obj_dict["META"][attr]
+        return super().__getstate__()
```

Comment:
This should be called first, i.e.
```suggestion
    def __getstate__(self):
        state = super().__getstate__()
        for attr in self.meta_non_picklable_items:
            if attr in state["META"]:
                del state["META"][attr]
        return state
```